src/convertToObject.js

```diff
@@ -6,7 +6,34 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.length) {
+    return {};
+  }
+
+  const noEmptyLines = sourceString
+    .split(';')
+    .map((line) => {
+      return line.trim();
+    })
+    .filter((item) => item.length > 0);
+
+  // trim
+  const noSpaces = noEmptyLines.map((item) => {
+    const [property, value] = item.split(':');
+
+    const newKey = property.trim();
+    const newValue = value.trim();
+
+    return `${newKey}:${newValue}`;
```

Comment:
[CODE KNOWLEDGE] - The intermediate array `noSpaces` is unnecessary. You can directly destructure and trim the properties and values within the `reduce` function, thus avoiding the creation of this intermediate array.